### PR TITLE
merge master (2ab7e6326) into go branch and fix build

### PR DESCRIPTION
**Pull request by jmhodges**
This merges master into the go branch and fixes up the build along the way. The build just needed the new filepaths to the C API header. And I ran `go generate` to get the latest protobuf.

Updates #10

### CODE REVIEW

**Comment by jmhodges** [comment]
Ah, the googlebot seems to not be able to check if commits listed are already in master and whitelist them from CLA checks. 

That's what all of these commits are except for the last two that are from me (who has a CLA)

Anything we can do to fix this?

**Comment by jmhodges** [comment]
Hey, would love to figure this one out! Lmk what I need to do!

**Comment by sbinet** [comment]
@jmhodges I believe you need to rebase `origin/go` on top of `master` and then `go_updates` on top of the rebased `go`.

see:
```sh
$> git fetch --all -p
$> git checkout go
$> git rebase master
$> git checkout go_updates
$> git rebase go
$> git l | head
* 9c5c181 (HEAD -> go_updates) run go generate on contrib/go
* 71c9cde some updates to header paths
* 9b5568d (go) remove non idiomatic NewGraphFromReader(reader, bool) and add NewGraphFromBuffer and New Graph
* f04f5df fix documentation link
* b6e9ec8 fix mispless and add minor corrections
* 5ddeb6e Go bindings tensors (#2479)
*   f71cc62 (origin/master, origin/HEAD, master) Merge pull request #4237 from caisq/branch_132376998
|\  
| *   fe19bcb Merge commit for internal changes
| |\  
|/ /  
| * 0949ece Starting the server before we (optionally) sleep for the case where no device_filters are set. Otherwise, the servers will wait to connect to each other before starting to train. We might as well start as soon as we can. Change: 132376998
| * cde9adb All other tensors from this list would be ignored, and instead the previous output of the decoder would be used. Change: 132374855
| * 6809660 Added a check for output_buffer_size <= 1 for ZlibOutputBuffer. Also adding some tests for Zlib compression reading / writing. Change: 132370925
| * d7bc08f Update generated Python Op docs. Change: 132368885
| * d41348b Cumulative bug fixes to get tf.learn working with open source TF. Change: 132368553
| * 6e369da Add new str_util::HumanReadableElapsedTime that formats a 'double seconds' value using appropriate units based on the magnitude of the time interval, and add tests for this. Change: 132365945
| * 51a8d2e Update ops-related pbtxt files. Change: 132363192
| * 73bc0a4 Disable "label" mode. Change: 132360445
| * 5c8bff0 tfdbg: Allow debug ops to watch uninitialized tensors
| * 264f2a7 Update generated Python Op docs. Change: 132348660
| * 7bad05c Delete deprecated TensorFlowDNN(Regressor|Classifier). Also add tf_random_seed to fix flakiness for the affected tests. Change: 132347779
```

hth.

**Comment by jmhodges** [comment]
Okay, rebasing `go` against `master` and then `go_updates` against `go` kept giving me merge conflicts when I would push here. I think because I wasn't able to push to `origin/go`.

I've just merged in master.

We could do this in two steps if someone who can push `origin/go` can step in. First step: rebase `go` against `master`, and then I can apply these last two commits from myself.

**Comment by martinwicke** [comment]
I rebased go onto master. You should be able to update your branch.

**Comment by jmhodges** [comment]
There we go!

Now, the next thing after this PR is the generated protobuf code. It doesn't currently build using `go install` or `go get` (which I think is a goal)

**Comment by martinwicke** [comment]
@jhseu PTAL.

File: tensorflow/contrib/go/tensorflow.swigcxx [hunk 1]
```diff
@@ -33,8 +32,6 @@ limitations under the License.
   $2 = $input.len;
 }
 
-%include "tensorflow/core/public/version.h"
-%include "tensorflow/core/public/tensor_c_api.h"
+%include "../../c/c_api.h"
```
Comment (jhseu) [inline tensorflow/contrib/go/tensorflow.swigcxx]: Relative includes are not used in our source code. Mind making it an absolute include?

**Comment by jhseu** [comment]
It's hard to review the generated protocol buffer code (from go generate), particularly the gzipped file descriptor stuff. Mind omitting those (I can run it manually if it's needed)? Alternatively we could abandon this in favor of the work-in-progress Go bindings, since they're different:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/go

There needs to be a few more additions to the C API to complete them. We want to build all new bindings on top of the C API (rather than SWIG to the C++ one).

**Comment by martinwicke** [comment]
@jmhodges friendly ping

**Comment by jmhodges** [comment]
I was informed that this wasn't likely to go in since the tensorflow team was working on a go wrapper already and that I wouldn't be able to do anything to help move their version along. This happened in the comments of #10.

Anyone else is free to take my work up

**Comment by martinwicke** [comment]
Ok. I will close this PR then.

**Pull request closed without merging**